### PR TITLE
boards: cleanup structure of wsn430 boards

### PR DESCRIPTION
**Pull request by haukepetersen**
~~rebased on #8058,~~ alternative to #7975

So when not taking the path of board revisions and just cleaning up the code tree a little bit, it seems to me we can achieve the same thing. For this PR just compare this commit from #7975 b64b8b04d00446613f6562f723df2cc9a0e41305 with the same done in this PR e02f33f2248efe7024752e4e8ed1d2e99556283e.

The changes look pretty similar, and I think we might as well get away with keeping a dedicated folder per board (variant), and just improving on the code sharing part... So I tend to close #7975 again and simply go with structural cleanups based on #8058.

### CODE REVIEW

**Comment by haukepetersen** [comment]
reabsed

**Comment by miri64** [comment]
Does it make sense to merge this? The WSN430 are soon to be deactivated I read on Tuesday on the IoT-Lab-Mailinglist (whose archive isn't public so I sadly can't link).

**Comment by PeterKietzmann** [comment]
Not on all sites. I think it is too early to remove that port completely.

**Comment by miri64** [comment]
Alright then. Hit the green ;-)

**Comment by PeterKietzmann** [comment]
Is this an ACK after reviewing the code?

**Comment by miri64** [comment]
> Is this an ACK after reviewing the code?

[You already ACK'd](https://github.com/RIOT-OS/RIOT/pull/8061#pullrequestreview-80247219). Is a second ACK necessary?

**Comment by PeterKietzmann** [comment]
I asked for one (above), so yes. I'm not so familiar with these platforms and their code base.

**Comment by miri64** [comment]
Ah sorry ^^" no I think I'm even less familiar than you.

**Comment by haukepetersen** [comment]
rebased.

@PeterKietzmann: as you have been running this code on the nodes, I think this makes you currently the only expert we have on these boards... So I don't think we will ever get someone else to give a better review then you did. So my proposal: @miri64 looks hard at the changes once more, and we merge this PR with both of your (more or less expert) verdicts.

File: boards/common/wsn430/board_init.c [hunk 1]
```diff
@@ -15,111 +15,102 @@
...
+    /* wait for XTAL to stabilize */
+    do {
+        /* clear oscillator fault flag */
+        IFG1 &= ~OFIFG;
+        /* time for flag to set */
+        for (uint16_t i = 0xFF; i > 0; i--) {}
+    }
+
+    /* oscillator fault flag still set? */
+    while ((IFG1 & OFIFG) != 0);
```
Comment (miri64) [inline boards/common/wsn430/board_init.c]: Why are you writing the while loop here without a body, but the for-loop above you are writing with (empty) body. In general loops without bodies can lead to cross-platform problems and our [coding conventions state they should be *with* a body](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#general)

File: boards/common/wsn430/board_init.c [hunk 2]
```diff
@@ -15,111 +15,102 @@
...
+            break;
+        case MCLK_4MHZ_SCLK_1MHZ:
+            BCSCTL2  = (SELM_2 | DIVM_1 | SELS | DIVS_3);
+            break;
+        case MCLK_8MHZ_SCLK_1MHZ:
+            BCSCTL2 = (SELM_2 | SELS | DIVS_3);
+            break;
+        default:
+            /* MCLK and SMCLK = XT2 (safe) */
+            BCSCTL2 = (SELM_2 | SELS);
```
Comment (miri64) [inline boards/common/wsn430/board_init.c]: Minor optimization proposal: why not make the `MCLK_...` defines names for theses combinations of flags?

```C
#define MCLK_2MHZ_SCLK_1MHZ (SELM_2 | DIVM_2 | SELS | DIVS_3)
#define MCLK_4MHZ_SCLK_1MHZ (SELM_2 | DIVM_1 | SELS | DIVS_3)
#define MCLK_8MHZ_SCLK_1MHZ (SELM_2 | SELS | DIVS_3)
#define MCLK_8MHZ_SCLK_8MHZ (SELM_2 | SELS)
```

then you can just `BSCTL2 = speed`. Or does that not make any sense?

File: boards/common/wsn430/board_init.c (hunk 1 above)
Comment (haukepetersen) [reply boards/common/wsn430/board_init.c]: japp, this is a mistake here, will fix.

File: boards/common/wsn430/board_init.c (hunk 2 above)
Comment (haukepetersen) [reply boards/common/wsn430/board_init.c]: yap, obviously :-) Should have seen this myself, thanks!

**Comment by haukepetersen** [comment]
addressed comments

File: boards/common/wsn430/board_init.c [hunk 3]
```diff
@@ -15,111 +15,90 @@
...
-{
-    irq_disable();
-    __msp430_cpu_speed = speed;
-    msp430_init_dco();
-    irq_enable();
-}
-
-/*---------------------------------------------------------------------------*/
-void msp430_init_dco(void)
+void msp430_init_cpuclk(uint32_t speed)
```
Comment (miri64) [inline boards/common/wsn430/board_init.c]: Also, BCSCTL2 seems to be `char` according to most sources I was able to find (and the header files of the toolchain). So why not use `uint8_t` for speed?

**Comment by haukepetersen** [comment]
fixed.

**Comment by miri64** [comment]
Tested on both wsn430_1.3b and wsn430_1.4 with `examples/default` and still works. Please squash.

**Comment by haukepetersen** [comment]
squashed.

**Comment by haukepetersen** [comment]
ups, broke the do-while loop when squashing, fixed now.

**Comment by miri64** [comment]
"Re"-tested on iotlab-m3 for both platforms. This time with the right commit ^^"

**Comment by haukepetersen** [comment]
thanks!

**Pull request merged**